ocp_resources/cron_job.py

```diff
@@ -0,0 +1,102 @@
+from ocp_resources.constants import TIMEOUT_4MINUTES
+from ocp_resources.resource import NamespacedResource
+
+
+class CronJob(NamespacedResource):
+    """
+    CronJob object. API reference:
+    https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/cron-job-v1/
+    """
+
+    api_group = NamespacedResource.ApiGroup.BATCH
+
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        client=None,
+        teardown=True,
+        schedule=None,
+        job_template=None,
+        timezone=None,
+        concurrency_policy=None,
+        starting_deadline_seconds=None,
+        suspend=None,
+        successful_jobs_history_limit=None,
+        failed_jobs_history_limit=None,
+        privileged_client=None,
+        yaml_file=None,
+        delete_timeout=TIMEOUT_4MINUTES,
+        **kwargs,
+    ):
+        """
+        Args:
+            name (str): Name of the endpoints resource
+            namespace (str): Namespace of endpoints resource
+            client: (DynamicClient): DynamicClient for api calls
+            teardown (bool): Indicates if the resource should be torn down at the end
+            privileged_client (DynamicClient): Privileged client for api calls
+            yaml_file (str): yaml file for the resource.
+            delete_timeout (int, optional): timeout associated with delete action
+            schedule (str): schedule of the cron job
+            job_template (JobTemplateSpec): describes the job that would be created when a cronjob would be executed
```

Comment:
`job_template` expect to get `JobTemplateSpec` object?

Reply:
As per the API doc, it is JobTemplateSpec type. https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/cron-job-v1/#CronJobSpec. I see both metadata and spec present,

Reply:
the docstring specify `JobTemplateSpec`, meaning you expect the user to sent `JobTemplateSpec` object here.

Reply:
I will change this to dict. It should be a dict